src/Magnum/Math/Random.h

```diff
@@ -0,0 +1,131 @@
+#ifndef Magnum_Math_Random_h
+#define Magnum_Math_Random_h
+
+// TO DO Licence things.
+
+#include <random>
+#include <chrono>
+#include "Magnum/Types.h"
+#include "Magnum/Math/Constants.h"
+#include "Magnum/Math/Vector2.h"
+#include "Magnum/Math/Vector3.h"
+#include "Magnum/Math/Quaternion.h"
+#include "Magnum/Math/Functions.h"
+
+namespace Magnum
+{
+namespace Math
+{
+
+namespace Implementation
+{
+static std::seed_seq seeds{{
+    static_cast<std::uintmax_t>(std::random_device{}()),
+    static_cast<std::uintmax_t>(std::chrono::steady_clock::now()
+                                    .time_since_epoch()
+                                    .count()),
+}};
+
+class RandomGenerator
+{
+public:
+    RandomGenerator() = delete;
+
+    template <typename T>
+    static typename std::enable_if<std::is_same<Int, T>::value, T>::type
+    generate(T start = -Magnum::Math::Constants<T>::inf(),
+             T end = Magnum::Math::Constants<T>::inf())
+    {
+        return std::uniform_int_distribution<T>{start, end}(generator());
+    }
+
+    template <typename T>
+    static typename std::enable_if<std::is_same<Float, T>::value, T>::type
+    generate(T start = -Magnum::Math::Constants<T>::inf(),
+             T end = Magnum::Math::Constants<T>::inf())
+    {
+        return std::uniform_real_distribution<T>{start, end}(generator());
+    }
+
+public:
+    static std::mt19937 &generator()
+    {
+        static std::mt19937 g{seeds};
```

Comment:
`static` is problematic because of thread safety issues, global constructors/destructors etc. What do you think about this?

```cpp
Math::RandomGenerator g; // the user is required to instantiate this first

auto a = Math::randomSignedScalar(g);
auto b = Math::randomUnitVector2(g);
```

The `RandomGenerator` thus wouldn't be in `Implementation` anymore, but will be exposed to the user, even though its only use would be to get passed to the `random*()` functions.

Reply:
I thought about it, and tried to keep minimalistic(however you are right for the thread safety), to keep it as a 1-step function. 

if you have no other proposal and think that this would be the best, then why not :)

Reply:
Not sure if best, but I didn't come up with anything better yet.

I think this should be also robust in case I decide to implement my own random generator later.

Reply:
Implemented some stuff. Wondering how you would find these. I feel quiteee doubtful.(Especially would love to hide the "generate" function;